Tests/NIOHTTP2Tests/ConfiguringPipelineTests.swift

```diff
@@ -105,4 +105,37 @@ class ConfiguringPipelineTests: XCTestCase {
         XCTAssertNoThrow(try self.clientChannel.finish())
         XCTAssertNoThrow(try self.serverChannel.finish())
     }
+
+    func testPreambleGetsWrittenOnce() throws {
+        // This test checks that the preamble sent by NIOHTTP2Handler is only written once. There are two paths
+        // to sending the preamble, in handlerAdded(context:) (if the channel is active) and in channelActive(context:),
+        // we want to hit both of these.
+        let connectionPromise: EventLoopPromise<Void> = self.serverChannel.eventLoop.makePromise()
+
+        // Register the callback on the promise so we run it as it succeeds (i.e. before fireChannelActive is called).
+        let handlerAdded = connectionPromise.futureResult.flatMap {
+            // Use .server to avoid sending client magic.
+            self.serverChannel.configureHTTP2Pipeline(mode: .server)
+        }
+
+        self.serverChannel.connect(to: try SocketAddress(unixDomainSocketPath: "/fake"), promise: connectionPromise)
+        XCTAssertNoThrow(try connectionPromise.futureResult.and(handlerAdded).wait())
```

Comment:
This `and` is unnecessary: the `flatMap` will cover it. You can just wait for `handlerAdded`.